pycbc/results/pygrb_postprocessing_utils.py

```diff
@@ -644,29 +645,28 @@ def load_time_slides(hdf_file_path):
 # =============================================================================
 # Function to load the segment dicitonary
 # =============================================================================
-def load_segment_dict(xml_file):
-    """Loads the segment dictionary """
-
-    # Get the mapping table
-    time_slide_map_table = \
-        load_xml_table(xml_file, glsctables.TimeSlideSegmentMapTable.tableName)
-    # Perhaps unnecessary as segment_def_id and time_slide_id seem to always
-    # be identical identical
-    segment_map = {
-        int(entry.segment_def_id): int(entry.time_slide_id)
-        for entry in time_slide_map_table
-    }
-    # Extract the segment table
-    segment_table = load_xml_table(
-        xml_file, glsctables.SegmentTable.tableName)
-    segment_dict = {}
-    for entry in segment_table:
-        curr_slid_id = segment_map[int(entry.segment_def_id)]
-        curr_seg = entry.get()
-        if curr_slid_id not in segment_dict:
-            segment_dict[curr_slid_id] = segments.segmentlist()
-        segment_dict[curr_slid_id].append(curr_seg)
-        segment_dict[curr_slid_id].coalesce()
+def load_segment_dict(hdf_file_path):
+    """
+    Loads the segment dictionary with the format
+    {slide_id: segmentlist(segments analyzed)}
+    """
+
+    # Long time slides will require mapping between slides and segments
+    hdf_file = h5py.File(hdf_file_path, 'r')
+    ifos = extract_ifos(hdf_file_path)
+    # Get slide IDs
+    slide_ids = numpy.arange(len(hdf_file[f'{ifos[0]}/search/time_slides']))
+    # Get segment start/end times
+    seg_starts = hdf_file['network/search/segments/start_time'][:]
+    seg_ends = hdf_file['network/search/segments/end_time'][:]
+    # Write list of segments
+    seg_list = segments.segmentlist()
+    for i in range(len(seg_starts)):
+        seg_list.append(segments.segment(seg_starts[i], seg_ends[i]))
+
+    # Write segment_dict in proper format
+    # At the moment of this comment, there is only one segment
```

Comment:
The segments in this context refer to the segments analyzed in `pycbc_multi_inspiral`. In the coh_ptf code the segments were defined in a more complicated way based on long time slides. We don't have that functionality for `pycbc_multi_inspiral` at this point, so the result of this function is a dictionary of one slide with the full analysis chunk (since it coalesces the smaller segments into one). Having multiple segments would imply a gap in the data in the current state of the code. 

I suppose the functions here are mostly redundant if we're not doing long slides. [This function](https://git.ligo.org/lscsoft/lalsuite/-/blob/master/lalapps/src/ring/coh_PTF_utils.c#L626) is where the original code set up this structure. The output gets written into the XML file which the `load_segment_dict` function here was meant to read. I attempted to emulate this to achieve the same result.